### ISSUE TITLE
Webeditor - Can't create branch with slash "/"

### ISSUE THREAD

**Issue by 0xbaadf00d** [gogits/gogs#3568]
- Gogs version (or commit ref): 
- Git version:  0.9.97.0830
- Operating system: Linux
- Database (use `[x]`):
  - [ ] PostgreSQL
  - [x] MySQL
  - [ ] SQLite
- Can you reproduce the bug at https://try.gogs.io:
  - [x] Yes : https://try.gogs.io/zero-x-baadf00d/test/_new/master/
  - [ ] No
  - [ ] Not relevant
- Log gist:
## Description

We can't create branch with slash "/" when using webeditor. Using slash is used on many projects to separating branch into category (ie:  feature/xxxx   hotfix/xxxxx)

```
New branch name must be valid alpha or numeric or dash(-_) or dot characters.
```

![image](https://cloud.githubusercontent.com/assets/1005086/18161873/27515b7c-7034-11e6-8505-a7e35a5830c1.png)

**Comment by LFlare** [comment]
Hi,

I think I've identified the areas in which I believe this bug may be caused in.

    https://github.com/gogits/gogs/blob/master/modules/auth/repo_form.go#L294
    https://github.com/gogits/gogs/blob/master/modules/auth/repo_form.go#L323
    https://github.com/gogits/gogs/blob/master/modules/auth/repo_form.go#L350

Upon further analysis, `AlphaDashDot` appears to correspond to the name testing over at

    https://github.com/go-macaron/binding/blob/master/binding.go#L246

As such, to fix this, the testing of a new branch name has to be fixed over at go-macaron/binding or fixed in-house...

**Comment by Unknwon** [comment]
Patch has pushed to fix this issue, please test on `release/0.11` branch or https://try.gogs.io.
